pkg/reconciler/events/auditlogs/auditlogs.go

```diff
@@ -64,6 +70,28 @@ func (c *Reconciler) ReconcileKind(ctx context.Context, s *v1alpha1.CloudAuditLo
 	s.Status.InitializeConditions()
 	s.Status.ObservedGeneration = s.Generation
 
+	// If GCP ServiceAccount is provided, configure workload identity.
+	if s.Spec.ServiceAccount != nil {
```

Comment:
It seems that we are duplicating this code all over, same with the finalizer logic..
One possibility could be to create a workloadIdentity reconciler, similar to what we have for PubSub base, and put everything there...  
say, inside `reconciler/identity`, you can create an IdentityReconciler, which has two public methods: ReconcileWorkloadIdentity and DeleteWorkloadIdentity... similar to https://github.com/google/knative-gcp/blob/master/pkg/reconciler/pubsub/reconciler.go.... 

Instead of Pubsubable, maybe you can create a new interface under the duck folder called `Identifiable` or something similar, which is OwnerReferable (see Pubsubable) and has a GetIdentity method... Each of the types should implement Identifiable, i.e., will have the GetIdentity method, which will just return `spec.serviceAccount`... 

I think with that you might be able to reduce a lot of this boilerplate code... And you can move all the wi pieces within that package, instead of pubsub? Does it make sense? WDTY?

Reply:
I agree with that. I am also worried about the duplication of the code. Thank you for pointing out the way to further reduce that!